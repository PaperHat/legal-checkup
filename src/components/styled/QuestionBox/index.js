import styled from 'styled-components'

const QuestionBox = styled.div`
  width: 260px;
  height: 120px;
  font-family: Helvetica;
  font-size: 25px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
`
QuestionBox.displayName = 'StyledQuestionBox'

export default QuestionBox
