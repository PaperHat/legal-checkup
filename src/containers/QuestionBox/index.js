import StyledQuestionBox from '@components/styled/QuestionBox'
import { connect } from 'react-redux'
import { getActiveQuestion } from '@state/selectors'

function mapStateToProps (state) {
  const { body: children } = getActiveQuestion(state)

  return { children }
}

export default connect(mapStateToProps)(StyledQuestionBox)
