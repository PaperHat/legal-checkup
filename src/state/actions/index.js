import {
  CHECKUP_COMPLETE,
  ERROR_QUESTION_INDEX_OUT_OF_BOUNDS,
  NEXT_QUESTION_ACTIVATED,
  NEXT_QUESTION_REQUESTED,
  PREVIOUS_QUESTION_ACTIVATED,
  PREVIOUS_QUESTION_REQUESTED,
  QUESTION_REQUESTED,
  REDO_CHECKUP_CLICKED,
  REQUESTED_QUESTION_ACTIVATED,
  USER_RESPONDED_WITH_NO,
  USER_RESPONDED_WITH_NOT_SURE,
  USER_RESPONDED_WITH_YES
} from '../constants'

// All questions have been answered - triggers the results page in the checkupComplete epic
export function checkupComplete () {
  return {
    type: CHECKUP_COMPLETE
  }
}

// The question index requested is out of bounds (< 0 or more than the current responses + 1)
export function errorQuestionIndexOutOfBounds (questionIndexRequested) {
  return {
    type: ERROR_QUESTION_INDEX_OUT_OF_BOUNDS,
    payload: {
      questionIndexRequested
    }
  }
}

// Move to the next question
export function nextQuestionActivated (activeQuestionIndex) {
  return {
    type: NEXT_QUESTION_ACTIVATED,
    payload: {
      activeQuestionIndex
    }
  }
}

// Request the next question - whether we move or error depends on the questionRequested epic
export function nextQuestionRequested () {
  return {
    type: NEXT_QUESTION_REQUESTED
  }
}

// Move to the previous question
export function previousQuestionActivated (activeQuestionIndex) {
  return {
    type: PREVIOUS_QUESTION_ACTIVATED,
    payload: {
      activeQuestionIndex
    }
  }
}

// Request the previous question - whether we move or error depends on the questionRequested epic
export function previousQuestionRequested () {
  return {
    type: PREVIOUS_QUESTION_REQUESTED
  }
}

// Request a specific question - whether we move or error depends on the questionRequested epic
export function questionRequested (questionIndex) {
  return {
    type: QUESTION_REQUESTED,
    payload: {
      questionIndex
    }
  }
}

// Move to the specified question
export function requestedQuestionActivated (activeQuestionIndex) {
  return {
    type: REQUESTED_QUESTION_ACTIVATED,
    payload: {
      activeQuestionIndex
    }
  }
}

// The user gave a NO response
export function userRespondedWithNo () {
  return {
    type: USER_RESPONDED_WITH_NO
  }
}

// The user gave a NOT SURE response
export function userRespondedWithNotSure () {
  return {
    type: USER_RESPONDED_WITH_NOT_SURE
  }
}

// The user gave a YES response
export function userRespondedWithYes () {
  return {
    type: USER_RESPONDED_WITH_YES
  }
}

// The user wants to redo the checkup
export function redoCheckupClicked () {
  return {
    type: REDO_CHECKUP_CLICKED
  }
}
