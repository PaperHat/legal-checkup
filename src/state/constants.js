// Actions
// These have been 'namespaced' to the Checkup part of the app
// We will eventually have many more actions (e.g., admin actions), so it will be nice
// to keep them all straight
// Note that these are easy to read and understand when you see them in the Redux DevTools
export const CHECKUP_COMPLETE = '@checkup/CHECKUP_COMPLETE'
export const ERROR_QUESTION_INDEX_OUT_OF_BOUNDS =
  '@checkup/ERROR_QUESTION_INDEX_OUT_OF_BOUNDS'
export const NEXT_QUESTION_ACTIVATED = '@checkup/NEXT_QUESTION_ACTIVATED'
export const NEXT_QUESTION_REQUESTED = '@checkup/NEXT_QUESTION_REQUESTED'
export const PREVIOUS_QUESTION_ACTIVATED =
  '@checkup/PREVIOUS_QUESTION_ACTIVATED'
export const PREVIOUS_QUESTION_REQUESTED =
  '@checkup/PREVIOUS_QUESTION_REQUESTED'
export const QUESTION_REQUESTED = '@checkup/QUESTION_REQUESTED'
export const REQUESTED_QUESTION_ACTIVATED =
  '@checkup/REQUESTED_QUESTION_ACTIVATED'
export const USER_RESPONDED_WITH_NO = '@checkup/USER_RESPONDED_WITH_NO'
export const USER_RESPONDED_WITH_NOT_SURE =
  '@checkup/USER_RESPONDED_WITH_NOT_SURE'
export const USER_RESPONDED_WITH_YES = '@checkup/USER_RESPONDED_WITH_YES'
export const QUESTION_SELECTED = '@checkup/QUESTION_SELECTED' // mine
// Response enumeration
export const NO = 'No'
export const NOT_SURE = 'Not sure'
export const YES = 'Yes'

// Page paths
// Always better to use constants for values used in more than one place in the code --
// It ensures that you don't have different values, which causes painful, hard to find bugs
export const HOME_PATH = '/'
export const CHECKUP_PATH = '/checkup'
export const PRIVACY_POLICY_PATH = '/privacy-policy'
export const RESULTS_PATH = '/results'
export const TERMS_OF_USE_PATH = '/terms-of-use'
