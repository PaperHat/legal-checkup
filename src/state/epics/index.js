import { combineEpics } from 'redux-observable'

import rerouteOnThirdTest from './rerouteOnThirdTest'

export default combineEpics(rerouteOnThirdTest)
