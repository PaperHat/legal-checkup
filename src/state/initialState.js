import { graphql } from 'gatsby'
import { isEmpty } from 'ramda'

const getInitialState = () => {
  const query = graphql`
    {
      allGoogleSheetQuestionsRow {
        edges {
          node {
            id
            status
            topic
            questiontext
            helptext
            resultstext
            resourcelink
          }
        }
      }
    }
  `

  return isEmpty(query) ? 'Nothing returned' : query
}

// Initial state (temporary)
const initialState = {
  query: getInitialState(),
  checkup: {
    activeQuestionIndex: 0,
    topics: [
      {
        id: '4DDB61C7677B4BA2813D2CAE98EF51D5',
        name: 'Money Troubles',
        questions: [
          {
            body: 'Is anyone chasing you for money?',
            help: '',
            results: 'The fact that someone is chasing you for money',
            id: '759CDC30C25B489A9EB71B7E859F2DD7',
            resource: 'credit and debt',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-13-credit-and-debt/what-this-chapter-covers-chapter-13/'
          },
          {
            body: 'Do you have payments due or unpaid accounts?',
            help:
              'phone / Work and Income / Cash converters / power bills / car loan / gyms',
            results: 'Your unpaid accounts or due payments',
            id: '8A9B3C23BA41440187B0B9F9EB0D8400',
            resource: 'credit and debt',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-13-credit-and-debt/what-this-chapter-covers-chapter-13/'
          },
          {
            body:
              "Have you recently had any problems with a product or item you've bought?",
            help: '',
            results:
              'The problem you had with a product or item you brought recently',
            id: 'A7D51C38F34A4EA1919C74013E703C39',
            resource: 'consumer protection',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-12-consumer-protection/what-this-chapter-covers-chapter-12/'
          },
          {
            body:
              'Have you recently had any problems with a service you’ve bought?',
            help: '',
            results:
              'The problem you had recently had with a service you have brought',
            id: '5DBE75B1D33D43DD8AC4E00730B5BD3D',
            resource: 'consumer protection',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-12-consumer-protection/what-this-chapter-covers-chapter-12/'
          },
          {
            body: 'Do you have any unpaid fines?',
            help: '',
            results: 'Your unpaid fines',
            id: 'E2C5A2E126474F56B76719EBC818686E',
            resource: 'court fines',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-32-court-fines/what-this-chapter-covers-chapter-32/'
          }
        ]
      },
      {
        id: '6A83066039A94C1088BBC759E66CB2E7',
        name: 'Housing',
        questions: [
          {
            body: 'Have you recently received an eviction notice?',
            help: '',
            results: 'Your eviction notice',
            id: '9FC65A2D929B4A7DB86BE5417A190928',
            resource: 'tenancy and housing',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-15-tenancy-and-housing/what-this-chapter-covers-chapter-15/'
          },
          {
            body: 'Has your landlord failed to lodge and/or return your bond?',
            help: '',
            results: 'That your bond has not been lodged or returned',
            id: 'C3D47C40225E4AE08C1512B79603ADB5',
            resource: 'bond and tenancy',
            url:
              'https://www.tenancy.govt.nz/about-tenancy-services/contact-us/'
          },
          {
            body: 'Are there any belongings you have been unable to pick up?',
            help: '',
            results: 'Your trouble collecting your belongings',
            id: '3BB7BDCBE09B48B99BFD74EC6A4466C0',
            resource: 'tenancy and housing',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-15-tenancy-and-housing/what-this-chapter-covers-chapter-15/'
          },
          {
            body: 'Do you have any unpaid rent?',
            help: '',
            results: 'Your unpaid rent',
            id: '86A9B3EDF65043DE92503EFD8B530475',
            resource: 'tenancy and housing',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-15-tenancy-and-housing/what-this-chapter-covers-chapter-15/'
          },
          {
            body: 'Do you have any problems with your landlord or flatmates?',
            help: '',
            results: 'Your problems with your landlord and/or flatmates',
            id: '6AA1854FF9354B95968BBA34B4F92F6E',
            resource: 'tenancy and housing',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-15-tenancy-and-housing/what-this-chapter-covers-chapter-15/'
          },
          {
            body: 'Do you have any problems with your neighbours?',
            help: '',
            results: 'Problems with your neighbours',
            id: '288852DF29C440259BC9C1FA141821D4',
            resource: 'neighbours',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-11-neighbourhood-life/what-this-chapter-covers-chapter-11/'
          },
          {
            body:
              'Are you having difficulties applying for government or community housing?',
            help: '',
            results:
              'Your difficulties with applying for government or community housing',
            id: '64BC22049E5A4FEEAA51505CF79C1571',
            resource: 'community and government housing',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-15-tenancy-and-housing/social-housing-tenants-in-state-and-community-housing-chapter-15/'
          }
        ]
      },
      {
        id: 'C0501ED83A8D4A42A1FA3E6BD7ADFC6F',
        name: 'Crime',
        questions: [
          {
            body: 'Are you scheduled to appear in court for any reason?',
            help: '',
            results: 'Your court appearance',
            id: '39BB22A504F445FEAB5EA0800B6EABA3',
            resource: 'criminal law',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-34-the-criminal-courts/what-this-chapter-covers-chapter-34/'
          },
          {
            body:
              'Have you or a family member had contact with the police you would like to discuss?',
            help: '',
            results: "Your, or your family member's contact with the police",
            id: '2D525EC086934DF0B82B8D01B7AF4E77',
            resource: 'criminal law',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-34-the-criminal-courts/what-this-chapter-covers-chapter-34/'
          }
        ]
      },
      {
        id: '07213088A2B44D4B982F6FF8A6C17407',
        name: 'Work and Income',
        questions: [
          {
            body: 'Are you having problems with Work and Income?',
            help: '',
            results: 'Your problems with Work and Income',
            id: '87DA47D8ACD14C89948A1513C91A8271',
            resource: 'dealing with Work and Income',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-7-dealing-with-work-and-income/what-this-chapter-covers-chapter-7/'
          },
          {
            body:
              'Would you like to change or review any of your Work and Income arrangements?',
            help: '',
            results: 'Reviewing or changing your Work and Income benefits',
            id: 'EF232E20AC6140A2830E3800954FDD61',
            resource: 'dealing with Work and Income',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-7-dealing-with-work-and-income/what-this-chapter-covers-chapter-7/'
          }
        ]
      },
      {
        id: '9A446019CF9842BB93D3097879973E25',
        name: 'Relationships and Family',
        questions: [
          {
            body:
              'Would you like to discuss the safety of yourself or your children?',
            help: '',
            results: 'The safety of yourself or your children',
            id: '4B5D2C495D7C4DC996E06E6505A6AAA2',
            resource: 'domestic violence',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-29-domestic-violence/what-this-chapter-covers-chapter-29/'
          },
          {
            body:
              'Would you like to discuss arrangements, plans or orders about children?',
            help:
              'who the child/ren live with, spend time with, or your involvement in major long-term issues',
            results: 'Arrangements, plans, or orders about your children',
            id: '522575FC96D443C294938C75EBEF4B45',
            resource: 'Parents, guardians and caregivers',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-27-parents-guardians-and-caregivers/what-this-chapter-covers-chapter-27/'
          },
          {
            body:
              'Do you have any concerns about Child Youth and Family and your whānau?',
            help: '',
            results: 'Concerns about Child Youth and Family and your whānau',
            id: '9C11FDA3C57543158261F434C2EDC88B',
            resource: 'Child Youth and Family',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-28-dealing-with-child-youth-and-family/what-this-chapter-covers-chapter-28/'
          },
          {
            body:
              'Would you like to discuss your rights in any personal relationship, including divorce or separation?',
            help: '',
            results:
              'Your rights in a personal relationship (including divorce and separation)',
            id: '30385CC6362443A4BA83DF081CD3D7E9',
            resource: 'personal relationships and the law',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-26-adult-relationships/what-this-chapter-covers-chapter-26/'
          },
          {
            body:
              'Concerns about your experiences as a child while you were in the care of adults or institutions?',
            help: '',
            results: 'Your experiences as a child',
            id: 'F6044D7F0A7E4F81B755D0099EE9FD6B',
            resource: '',
            url: ''
          }
        ]
      },
      {
        id: 'B4EC55D2014740F88AD3E3F94371A3F5',
        name: 'Employment',
        questions: [
          {
            body: 'Are you having any problems at work?',
            help: '',
            results: 'Your problems at work',
            id: 'F3290A11500A4EABAD05DC0F3005F4AF',
            resource: 'resolving employment problems',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-18-resolving-employment-problems/what-this-chapter-covers-chapter-18/'
          },
          {
            body:
              'Are you worried about getting fired, laid off, or having your hours cut?',
            help: '',
            results:
              'Your concerns about losing your job or having your hours cut',
            id: 'BC60D4FA6BC548D9A28F14261C883E87',
            resource: 'resolving employment problems',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-18-resolving-employment-problems/what-this-chapter-covers-chapter-18/'
          },
          {
            body: 'Does your employer owe you money?',
            help: '',
            results: 'Your employer owes you money',
            id: '11BF1DB8DC6941D7B9057B46154DA6DA',
            resource: 'resolving employment problems',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-18-resolving-employment-problems/what-this-chapter-covers-chapter-18/'
          },
          {
            body:
              'Do you feel like you have been treated unfairly due to your gender, ethnicity, religion etc?',
            help: '',
            results: 'Discrimination at work',
            id: '2CDB3B20E34E48E29091055A678CB400',
            resource: 'resolving employment problems',
            url:
              'http://communitylaw.org.nz/community-law-manual/chapter-18-resolving-employment-problems/what-this-chapter-covers-chapter-18/'
          }
        ]
      },
      {
        id: '6FC5D30617CE43D6A0EB05D0C4243037',
        name: 'Other',
        questions: [
          {
            body:
              'Are you worried about how your child is being treated at school?',
            help: '',
            results: 'Concerns about how your child is being treated at school',
            id: '692D1218BF3E43489B7F8DF76A679713',
            resource: 'student rights',
            url: 'http://www.wclc.org.nz/student-rights-service/'
          },
          {
            body:
              'Do you have concerns around your immigration status in New Zealand?',
            help: '',
            results: 'Immigration status',
            id: '6729B7CD5B6B425FB782260BEEAE47F4',
            resource: '',
            url: ''
          },
          {
            body:
              'Are you having trouble bringing family members to New Zealand?',
            help: '',
            results: 'Trouble bringing family members to New Zealand',
            id: '79591D25C03D4AA7912E1366DD282586',
            resource: '',
            url: ''
          }
        ]
      }
    ],
    responses: []
  }
}

export default initialState
