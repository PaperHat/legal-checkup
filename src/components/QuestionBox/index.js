import React from 'react'
import { StyledQuestionBox } from '../styled'

export default function QuestionBox ({ questions, activeQuestion }) {
  return <StyledQuestionBox>{questions[activeQuestion].body}</StyledQuestionBox>
}
/* check where to pass question box component to be rendered
  it goes through the home page or checkup page?
  */
