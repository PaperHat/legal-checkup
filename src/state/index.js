import { nextQuestion, previousQuestion, testCount } from './actions'
import {
  INITIAL_STATE,
  NEXT_QUESTION,
  PREVIOUS_QUESTION,
  TEST_COUNT
} from './constants'
import rootEpic from './epics'
import { rootReducer } from './reducers'
import { getActiveQuestion, getCount, getTestCount } from './selectors'

export {
  getActiveQuestion,
  getCount,
  getTestCount,
  INITIAL_STATE,
  NEXT_QUESTION,
  nextQuestion,
  PREVIOUS_QUESTION,
  previousQuestion,
  rootEpic,
  rootReducer,
  TEST_COUNT,
  testCount
}
