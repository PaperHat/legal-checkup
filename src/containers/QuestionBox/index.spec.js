import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import { state } from '../../state/fixtures'

import QuestionBox from '.'

const mockStore = configureStore()

const store = mockStore(state)

describe('containers:QuestionBox', () => {
  it('renders the QuestionBox', () => {
    expect(toJson(shallow(<QuestionBox store={store} />))).toMatchSnapshot()
  })

  it('should map activeQuestion to props', () => {
    const wrapper = shallow(<QuestionBox store={store} />)
    expect(wrapper.props().children).toBe('Is anyone chasing you for money?')
  })
})
