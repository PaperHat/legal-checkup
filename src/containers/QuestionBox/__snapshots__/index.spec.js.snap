// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers:QuestionBox renders the QuestionBox 1`] = `
<StyledQuestionBox
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
>
  Is anyone chasing you for money?
</StyledQuestionBox>
`;
